lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java

```diff
@@ -213,6 +214,14 @@ private void flush()
         }
         List<BufferData> bufferDataList = builder.build();
 
+        if (rows == 0) {
+            // Avoid writing empty row groups as these are ignored by the reader
+            verify(
+                    bufferDataList.stream().allMatch(buffer -> buffer.getData().size() == 0),
```

Comment:
How many buffers there can be? Stream is potentially slow

Reply:
It's equal to number of columns, this will be called once per file so it shouldn't cause perf problems